templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -364,76 +349,97 @@ contents:
       echo "OVS configuration successfully reverted"
     }
 
-    # Reloads NetworkManager if any configuration change was done
-    reload_nm() {
+    # Reloads NM NetworkManager profiles if any configuration change was done.
+    # Accepts a list of devices that should be re-connect after reload.
+    reload_profiles_nm() {
       if [ $nm_config_changed -eq 0 ]; then
         # no config was changed, no need to reload
         return
       fi
-      nm_config_changed=0
-      
-      echo "Reloading NetworkManager after configuration changes..."
 
-      # recycle network, so that existing profiles and priorities are re-evaluated
-      nmcli network off
-
-      # wait for no devices to show as connected
-      echo "Waiting for devices to disconnect..."
-      if ! timeout 60 bash -c "while nmcli -g DEVICE,STATE d | grep -v :unmanaged; do sleep 5; done"; then
-        echo "Warning: NetworkManager did not disconnect all devices"
-      fi
-      
-      # reload profiles and set networking back on
+      # reload profiles
       nmcli connection reload
-      nmcli network on
-      
-      # restart NetworkManager so that we can wait on `nm-online -s`
-      systemctl restart NetworkManager
 
-      # Wait until all profiles auto-connect
-      echo "Waiting for profiles to activate..."
-      if nm-online -s -t 60; then
-        echo "NetworkManager has activated all suitable profiles after reload"
-      else
-        echo "Warning: NetworkManager has not activated all suitable profiles after reload"
-      fi
+      # precautionary sleep of 10s (default timeout of NM to bring down devices)
+      sleep 10
```

Comment:
this sleep is kind of scary... why is nmcli conn reload taking down devices? What happens if we dont sleep here? try to do nmcli device connect too early?

Reply:
`nmcli conn reload` only takes down devices that have no profile. In our case, this will happen to the physical device we had in br-ex on rollback, once we remove `ovs-if-phys0`, if it was originally activated through an ephemeral generated profile.

With this sleep I am trying to prevent as much as possible that we don't find,  after reload,  a device that's is to be disconnected because of the previous reason in `disconnecting` state instead of `disconnected`. The default timeout for `nmcli disconnec <dev>` is 10 seconds so using it here.